lib/components/AlbumScreen/playlist_edit_dialog.dart

```diff
@@ -99,19 +95,13 @@ class _PlaylistNameEditDialogState extends State<PlaylistNameEditDialog> {
     );
   }
 
-  Future<void> _fetchPublicVisibility() async {
-    final dynamic resultPlaylist = await _jellyfinApiHelper.getPlaylist(_id!);
-    final Map<String, dynamic> data =
-        jsonDecode(resultPlaylist as String) as Map<String, dynamic>;
-    // OpenAccess determines whether a playlist is publicly visible
-    try {
-      _publicVisibility = data['OpenAccess'] as bool;
-    } catch (e) {
-      // Just a silly idea to maybe make it a bit future proof
-      _publicVisibility = data['isPublic'] as bool;
-    }
-    print(_publicVisibility);
-  }
+Future<void> _fetchPublicVisibility() async {
+  if (_publicVisibility != null) return;
+  final resultPlaylist = await _jellyfinApiHelper.getPlaylist(widget.playlist.id!);
+  _publicVisibility = resultPlaylist['OpenAccess'] as bool;  
+  setState((){});
```

Comment:
If you're modifying a variable that is part of the state, and want the widget to refresh to show the new state, then you should always put the modifications into the the `setState()` callback, like so:

```dart
setState((){
  _publicVisibility = resultPlaylist['OpenAccess'] as bool;
});
```

I'll change it for you before merging :)